### PR TITLE
Storages: log region applied status when write

### PR DESCRIPTION
**Pull request by JinheLin**
### What problem does this PR solve?

Issue Number: ref #8864

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
```
[2024/04/11 16:53:47.130 +08:00] [DEBUG] [DeltaMergeStore.cpp:567] ["region_id: 924121, applied_index: 1569337, record_count: 10, versions: {449006961195221023}"] [source="keyspace=4294967295 table_id=3009"] [thread_id=207]
```
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Core/Block.h [hunk 1]
```diff
@@ -52,6 +53,10 @@ class Block
     // Only used for calculating MVCC-bitmap-filter.
     ColumnPtr segment_row_id_col;
 
+    // `region_id` is use to trace which region this block belongs to when writing.
+    // Only use for logging and debugging.
+    RegionID region_id = 0;
```
Comment (JaySon-Huang) [inline dbms/src/Core/Block.h:58]: I think it is not a good idea to add a `RegionID` to `Block` that is being widely used everywhere in the code.

Comment (CalvinNeo) [reply dbms/src/Core/Block.h:58]: I agree with @JaySon-Huang , Region and Segments are not always strictly mapped

File: dbms/src/Storages/DeltaMerge/DeltaMergeStore.cpp [hunk 2]
```diff
@@ -533,7 +533,11 @@ Block DeltaMergeStore::addExtraColumnIfNeed(
     return std::move(block);
 }
 
-DM::WriteResult DeltaMergeStore::write(const Context & db_context, const DB::Settings & db_settings, Block & block)
+DM::WriteResult DeltaMergeStore::write(
+    const Context & db_context,
+    const DB::Settings & db_settings,
+    Block & block,
+    RegionID region_id)
```
Comment (CalvinNeo) [inline dbms/src/Storages/DeltaMerge/DeltaMergeStore.cpp:540]: We also need to print out `applied_index`. So better changed to a context struct.

File: dbms/src/Storages/DeltaMerge/DeltaMergeStore.cpp [hunk 3]
```diff
@@ -554,7 +558,7 @@ DM::WriteResult DeltaMergeStore::write(const Context & db_context, const DB::Set
         {
             dedup_ver.insert(v);
         }
-        LOG_DEBUG(log, "Record count: {}, Versions: {}", block.rows(), dedup_ver);
+        LOG_DEBUG(log, "RegionID: {}, Record count: {}, Versions: {}", region_id, block.rows(), dedup_ver);
```
Comment (CalvinNeo) [inline dbms/src/Storages/DeltaMerge/DeltaMergeStore.cpp:561]:
```suggestion
        LOG_DEBUG(log, "region_id={}, record count={}, versions={}", region_id, block.rows(), dedup_ver);
```

**Review by CalvinNeo — APPROVED** [review]
lgtm

**Review by JaySon-Huang — APPROVED** [review]

**Pull request merged**